frontend/packages/noobaa-storage-plugin/src/components/object-bucket-claim-page/secret.tsx

```diff
@@ -1,36 +1,79 @@
 import * as React from 'react';
 import * as _ from 'lodash';
-import { Base64 } from 'js-base64';
-import { SecretData } from '@console/internal/components/configmap-and-secret-data';
+import { SecretValue } from '@console/internal/components/configmap-and-secret-data';
 import { ConfigMapModel, SecretModel } from '@console/internal/models';
 import { k8sGet, K8sResourceKind } from '@console/internal/module/k8s';
 import { getName, getNamespace } from '@console/shared';
+import { SectionHeading, EmptyBox } from '@console/internal/components/utils';
+import { Button } from '@patternfly/react-core';
+import { EyeSlashIcon, EyeIcon } from '@patternfly/react-icons';
 import { isBound } from '../../utils';
 
 export const GetSecret: React.FC<GetSecretProps> = ({ obj }) => {
-  const secretData = {
-    BUCKET_NAME: Base64.encode(getName(obj)),
-    ACCESS_KEY: Base64.encode('loading..'),
-    SECRET_KEY: Base64.encode('loading..'),
-    ENDPOINT: Base64.encode('http://loading.com'),
-  };
+  const [reveal, setReveal] = React.useState(false);
+  const [secretData, setSecretData] = React.useState([]);
+  const bucketName = getName(obj);
   if (isBound(obj)) {
     const secret = k8sGet(SecretModel, getName(obj), getNamespace(obj));
     const configMap = k8sGet(ConfigMapModel, getName(obj), getNamespace(obj));
     Promise.all([secret, configMap])
       .then((data) => {
-        secretData.ACCESS_KEY = Base64.encode(_.get(data[0], 'data.AWS_ACCESS_KEY_ID'));
-        secretData.SECRET_KEY = Base64.encode(_.get(data[0], 'data.AWS_SECRET_ACCESS_KEY'));
         const endpoint = `${_.get(data[1], 'data.BUCKET_HOST')}:${_.get(
           data[1],
           'data.BUCKET_PORT',
         )}`;
-        secretData.ENDPOINT = Base64.encode(endpoint);
+        const accessKey = _.get(data[0], 'data.AWS_ACCESS_KEY_ID');
+        const secretKey = _.get(data[0], 'data.AWS_SECRET_ACCESS_KEY');
+        const secretValues = [
+          { Endpoint: endpoint },
+          { 'Bucket Name': bucketName },
```

Comment:
```suggestion
          { name: 'Bucket Name', value: bucketName },
```

is cleaner. Then you don't need `Object.keys(datum)[0]` below